src/Microsoft.OData.Client/EntityTracker.cs

```diff
@@ -576,4 +576,12 @@ private void ValidateDuplicateIdentity(Uri identity, EntityDescriptor descriptor
             }
         }
     }
+
+    public class CustomComparer<TObject> : IEqualityComparer<TObject> where TObject : class

```

Comment:
I'd probably name this `EntityEqualityComparer` or something more contextual rather than just `CustomComparer`